nvflare/recipe/fedavg.py

```diff
@@ -0,0 +1,362 @@
+    """
+
+    def __init__(
+        self,
+        *,
+        name: str = "fedavg",
+        initial_model: Any = None,
+        initial_params: Optional[dict] = None,
+        min_clients: int,
+        num_rounds: int = 2,
+        train_script: str,
+        train_args: Union[str, Dict[str, str]] = "",
+        aggregator: Optional[Aggregator] = None,
+        aggregator_data_kind: Optional[DataKind] = DataKind.WEIGHTS,
+        launch_external_process: bool = False,
+        command: str = "python3 -u",
+        framework: FrameworkType = FrameworkType.PYTORCH,
+        server_expected_format: ExchangeFormat = ExchangeFormat.NUMPY,
+        params_transfer_type: TransferType = TransferType.FULL,
+        model_persistor: Optional[ModelPersistor] = None,
+        model_locator: Optional[ModelLocator] = None,
+    ):
+        # Validate inputs internally
+        v = _FedAvgValidator(
+            name=name,
+            initial_model=initial_model,
+            initial_params=initial_params,
+            min_clients=min_clients,
+            num_rounds=num_rounds,
+            train_script=train_script,
+            train_args=train_args,
+            aggregator=aggregator,
+            aggregator_data_kind=aggregator_data_kind,
+            launch_external_process=launch_external_process,
+            command=command,
+            framework=framework,
+            server_expected_format=server_expected_format,
+            params_transfer_type=params_transfer_type,
+            model_persistor=model_persistor,
+            model_locator=model_locator,
+        )
+
+        self.name = v.name
+        self.initial_model = v.initial_model
+        self.initial_params = v.initial_params
+        self.min_clients = v.min_clients
+        self.num_rounds = v.num_rounds
+        self.train_script = v.train_script
+        self.train_args = v.train_args
+        self.aggregator = v.aggregator
+        self.aggregator_data_kind = v.aggregator_data_kind
+        self.launch_external_process = v.launch_external_process
+        self.command = v.command
+        self.framework = v.framework
+        self.server_expected_format = v.server_expected_format
+        self.params_transfer_type = v.params_transfer_type
+        self.model_persistor = v.model_persistor
+        self.model_locator = v.model_locator
+
+        # Validate that only one of initial_model or initial_params is provided
+        if self.initial_model is not None and self.initial_params is not None:
+            raise ValueError(
+                "Cannot provide both initial_model and initial_params. "
+                "Use initial_model for PyTorch/TensorFlow, initial_params for sklearn."
+            )
+
+        # For sklearn (RAW framework), we handle persistor differently
+        if self.framework == FrameworkType.RAW:
+            # Sklearn case - use JoblibModelParamPersistor
+            persistor = JoblibModelParamPersistor(initial_params=self.initial_params or {})
```

Comment:
can we push this / pass this in the sklearn/recipes/fedavg.py instead of doing it here in the base